### PR TITLE
Issue 674

### PR DESCRIPTION
**Pull request by alimand**
Fix issue 674 and issue 683:
Adding new dashboard in Grafana to monitor the status of elasticsearch health by using metrics:
  1. elasticsearch_indices_settings_stats_read_only_indices(add alert)
  2. elasticsearch_filesystem_data_available_bytes
  3. elasticsearch_filesystem_data_free_bytes
Adding disk space usage status moniring and alert in new dashboard in Grafana
Optimize container relationship in wis2box to make sure when machine restart wis2box could bahave well in 1-2 minutes.

### CODE REVIEW

File: docker-compose.monitoring.yml [hunk 1]
```diff
@@ -108,6 +125,14 @@ services:
     <<: *logging
   elasticsearch:
     <<: *logging
+    container_name: elasticsearch
```
Comment (maaikelimper) [inline docker-compose.monitoring.yml:128]: why did you add another "elasticsearch"-services into the docker-compose.monitoring.yml ? I think this is a mistake and should be removed

File: docker-compose.yml [hunk 2]
```diff
@@ -86,6 +87,8 @@ services:
       - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
       - cluster.name=es-wis2box
       - xpack.security.enabled=false
+    ports:
+      - "9200:9200"
```
Comment (maaikelimper) [inline docker-compose.yml:91]: This additional port-mapping should not be required, please remove it

File: docker-compose.yml [hunk 3]
```diff
@@ -147,6 +150,8 @@ services:
 
   wis2downloader:
     container_name: wis2downloader
+    image: wis2box_project-wis2downloader
```
Comment (maaikelimper) [inline docker-compose.yml:153]: why did you define the image name  to the downloader image ? should be removed

File: prometheus/prometheus.yml [hunk 4]
```diff
@@ -8,26 +8,42 @@ global:
...
+
 - job_name: wis2downloader
   metrics_path: /metrics
   scheme: http
   static_configs:
   - targets: ['wis2downloader:5000']
+
+- job_name: 'prometheus'
+  static_configs:
+  - targets: ['localhost:9090']
```
Comment (maaikelimper) [inline prometheus/prometheus.yml:39]: what does this job do ?

File: prometheus/prometheus.yml [hunk 5]
```diff
@@ -8,26 +8,42 @@ global:
...
+
+- job_name: 'node_exporter'
```
Comment (maaikelimper) [inline prometheus/prometheus.yml:41]: remove the node_exporter for now

File: docker-compose.monitoring.yml [hunk 6]
```diff
@@ -25,14 +25,16 @@ services:
...
     command: -config.file=/etc/loki/loki-config.yml
     volumes:
       - loki-data:/loki
       - ./loki/loki-config.yml:/etc/loki/loki-config.yml
-    networks:
-      vpcbr: # this is the place where we assign the static ipv4 address
-        ipv4_address: 10.5.0.2
-      default:
+
+  node_exporter:
```
Comment (maaikelimper) [inline docker-compose.monitoring.yml:34]: remove the node-exporter for now

**Review by maaikelimper — CHANGES_REQUESTED** [review]
Hi Peiwen, I see several changes in the code that I think are related to local development and should be removed. Could you please add a screenshot of the new dashboard that you made ?

**Comment by alimand** [comment]
> Hi Peiwen, I see several changes in the code that I think are related to local development and should be removed. Could you please add a screenshot of the new dashboard that you made ?

Yes, sure Maiike. Thanks for your correctness, let me change this part!

File: docker-compose.monitoring.yml (hunk 1 above)
Comment (alimand) [reply docker-compose.monitoring.yml:128]: Yes, thats an error! I removed it

**Comment by alimand** [comment]
> Hi Peiwen, I see several changes in the code that I think are related to local development and should be removed. Could you please add a screenshot of the new dashboard that you made ?

image
Of course, here is the screenshot of the new dashboard in Grafana

File: prometheus/prometheus.yml (hunk 4 above)
Comment (alimand) [reply prometheus/prometheus.yml:39]: I removed this part it's useless

**Comment by maaikelimper** [comment]
Hi Peiwen, thank your for the screenshot. 

I think you can improve this dashboard for the wis2box-users:

"alarmtracking dashboard" -> this is the title of the dashboard but you named the json file that defines it "metricsmonitoring.json" please be consistent and think of a proper title and filename , you should mention "Elasticsearch" if the metrics are based on Elasticsearch

"disk space usage" -> what is the unit ? the Y-axis shows 3 times 27 which is confusing , can you please improve this 

"count of indices that have read_only_allow_delete=true" -> it is sufficient to display the current value for this metric, anything above 0 should be displayed in red

What is the difference between "available space" and "free space" ? Are they both required for the user ? Please display bytes using a human-friendly unit (Grafana has built-in option to auto-convert bytes into kB/MB)

the screenshots shows a range of "last 5 minutes" please define a sensible default for the default time-range

finally think about the usage of colors in your dashboard, for example you can set thresholds in Grafana to make a bad value display as red, while good values can be displayed in green

**Comment by alimand** [comment]
> Hi Peiwen, thank your for the screenshot.
> 
> I think you can improve this dashboard for the wis2box-users:
> 
> "alarmtracking dashboard" -> this is the title of the dashboard but you named the json file that defines it "metricsmonitoring.json" please be consistent and think of a proper title and filename , you should mention "Elasticsearch" if the metrics are based on Elasticsearch
> 
> "disk space usage" -> what is the unit ? the Y-axis shows 3 times 27 which is confusing , can you please improve this
> 
> "count of indices that have read_only_allow_delete=true" -> it is sufficient to display the current value for this metric, anything above 0 should be displayed in red
> 
> What is the difference between "available space" and "free space" ? Are they both required for the user ? Please display bytes using a human-friendly unit (Grafana has built-in option to auto-convert bytes into kB/MB)
> 
> the screenshots shows a range of "last 5 minutes" please define a sensible default for the default time-range
> 
> finally think about the usage of colors in your dashboard, for example you can set thresholds in Grafana to make a bad value display as red, while good values can be displayed in green

Sure, thanks for your review Maaike, let me impove it based on the comments. Will provide new version soon.

**Comment by alimand** [comment]
Some updates based on suggestions and optimize:
Modify all the sugguestions part;
At each top of the dashboard, the alert list will be shown (only the alerting one)
When alert is triggered, notifications could be sent to personal emails
Test Showcase:
WechatIMG735
WechatIMG736
WechatIMG737
WechatIMG738
WechatIMG8601
WechatIMG8605

**Comment by alimand** [comment]
Current Alerts Overview:
    Purpose: Displays a list of currently active alerts.

Elasticsearch Disk Space Utilization:
    Purpose: This panel shows the percentage of disk space used on the Elasticsearch node. It is an important metric to monitor the available disk space and ensure that the Elasticsearch cluster has sufficient capacity.
    Alert Condition: Triggers if disk usage exceeds 80% for 5 minutes.

Available Disk Space (GB):
    Purpose: This panel shows the amount of free disk space available on the block device where Elasticsearch stores its data. Monitoring this helps prevent the disk from running out of space.
    Alert Condition: Triggers if free space falls below 10 GB for 5 minutes.

Elasticsearch Disk Usage Thresholds:
    Purpose: This panel shows the high and low disk watermarks configured for Elasticsearch in kilobytes (KB). These watermarks determine when Elasticsearch should start taking action to prevent nodes from running out of disk space, such as stopping new data allocations.
Disk Read and Write Activity:
    Purpose: This panel displays the rate of filesystem read and write operations per second. It helps to monitor how frequently data is being read from or written to the disk, which can be indicative of disk performance or potential bottlenecks.

Read-Only Indices Count:
    Purpose: This setting is often used to prevent further writes to indices that are running out of disk space.
    Alert Condition: Triggers if any read-only indices are detected (count > 0).

**Comment by alimand** [comment]
updates shown in issue-674-new branch
